core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/RemoteInsertOneResult.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2018-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.stitch.core.services.mongodb.remote;
+
+import org.bson.BsonValue;
+
+public class RemoteInsertOneResult {
+
+  private final BsonValue insertedId;
```

Comment:
[q] In general, why are these `BsonValue`s and not `ObjectId`s?

Reply:
Because an inserted id may be any bson value based on the collectible codec interface